File: CodeWars/JS/8kyu_is_the_string_upper_case.js
// Is the string uppercase?
// Task

// Add the isUpperCase method to String to see whether the string is ALL CAPS.
//  For example:

// "c".isUpperCase() == false
// "C".isUpperCase() == true
// "hello I AM DONALD".isUpperCase() == false
// "HELLO I AM DONALD".isUpperCase() == true
// "ACSKLDFJSgSKLDFJSKLDFJ".isUpperCase() == false
// "ACSKLDFJSGSKLDFJSKLDFJ".isUpperCase() == true

// In this Kata, a string is said to be in ALL CAPS whenever it does not
//  contain any lowercase letter so any string containing no letters at all
//   is trivially considered to be in ALL CAPS.

String.prototype.isUpperCase=function() {return this==this.toUpperCase()}

console.log("C".isUpperCase()) //== true)
console.log("hello I AM DONALD".isUpperCase()) //== false)
console.log("HELLO I AM DONALD".isUpperCase()) // == true)
console.log("ACSKLDFJSgSKLDFJSKLDFJ".isUpperCase()) // == false)
console.log("ACSKLDFJSGSKLDFJSKLDFJ".isUpperCase()) // == true)

File: CodeWars/python/8kyu_is_the_string_uppercase.py
# Is the string uppercase?
# Task

# Create a method is_uppercase() to see whether the string
#  is ALL CAPS. For example:

# is_uppercase("c") == False
# is_uppercase("C") == True
# is_uppercase("hello I AM DONALD") == False
# is_uppercase("HELLO I AM DONALD") == True
# is_uppercase("ACSKLDFJSgSKLDFJSKLDFJ") == False
# is_uppercase("ACSKLDFJSGSKLDFJSKLDFJ") == True


# In this Kata, a string is said to be in ALL CAPS whenever
#  it does not contain any lowercase letter so any string
#  containing no letters at all is trivially considered to
#  be in ALL CAPS.
# def is_uppercase(inp):
#     return all(map(lambda x: x.isupper() if x.isalpha() else True, inp))

def is_uppercase(inp):
    return inp.isupper()

print(is_uppercase("c"))  # False)
print(is_uppercase("C"))  # True)
print(is_uppercase("hello I AM DONALD"))  # False)
print(is_uppercase("HELLO I AM DONALD"))  # True)
